### PR TITLE
feat(GlobalHeader): add getGlobalHeaderHeight helper function

### PR DESCRIPTION
**Pull request by christian-dinh**
### Overview

~There are two different ways I tried to do this, both visible in two commits. the first (`with refs`) is fancier but probably unnecessarily so. It uses refs and `resize` listeners inside of `useLayoutEffect` to function agnostically from how the header is sized. The second (`without refs`) simply sets the height based on `useBreakpointAtOrAbove` and uses it for both the imperative handle and the sizing of the header itself. Both use `useImperativeHandle`, but the latter way seems better to me so that's what the main diff is. If you look at the first commit on its own and prefer it I am open to discussion! Also open to alternatives to an imperative handle if anyone has a better idea.~

Adds a ~hook~ helper function for getting the current height of the `GlobalHeader` for layouts that require that info. Includes docs and unit tests!

### PR Checklist

- [ ] Related to designs:
- [x] Related to JIRA ticket: [REACH-619](https://codecademy.atlassian.net/browse/REACH-619)
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change

### CODE REVIEW

File: packages/gamut-labs/src/experimental/GlobalHeader/index.tsx [hunk 1]
```diff
@@ -93,22 +94,25 @@ const getMobileAppHeaderItems = (
   }
 };
 
-export const GlobalHeader: React.FC<GlobalHeaderProps> = (props) => (
-  <>
-    <Box display={{ base: 'none', md: 'block' }} height="80">
```
Comment (christian-dinh) [inline packages/gamut-labs/src/experimental/GlobalHeader/index.tsx:98]: if there's a reason we did this instead of `useBreakpointAtOrAbove` that I didn't see let me know. I don't have anything against this approach, but switching to `useBreakpoint` let me use 1 ref instead of 2 in the `with refs` version and reduced repetition in the `without refs` version 🤷

Comment (jakemhiller) [reply packages/gamut-labs/src/experimental/GlobalHeader/index.tsx:98]: useBreakpointAtOrAbove uses JS to calculate breakpoints, the method here generates CSS using emotion, and is native to the Box component. Using the hook would have been unnecessary overhead unless you needed to use the current value elsewhere

Comment (christian-dinh) [reply packages/gamut-labs/src/experimental/GlobalHeader/index.tsx:98]: Oh yeah that makes sense. There isn't really a way around those performance implications in either version of this code. There are other ways to get the header height without an imperative handle (eg: `getElementsByTagName('header')` in static-sites) if this is not a worthwhile tradeoff.

Comment (rogervera) [reply packages/gamut-labs/src/experimental/GlobalHeader/index.tsx:98]: Echoing Jake, I suggested this to @katiezutter when she did some refactoring as part of https://github.com/Codecademy/client-modules/pull/1398#pullrequestreview-589715971. 

Historically, @judahtanthony wrote the original `breakpointHooks` hooks in the monolith, before we ported them to `gamut-labs`. This was all _before_ Emotion.

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@10.0.12-alpha.dc1f75.0<br/>@codecademy/styleguide@27.2.2-alpha.dc1f75.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://602eb184b96f44013b735f68--gamut-preview.netlify.app>https://602eb184b96f44013b735f68--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/602eb184b96f44013b735f68>Deploy Logs</a></p>

**Comment by christian-dinh** [comment]
this is still a draft, final version will have unit tests if this looks like a reasonable add

File: packages/gamut-labs/src/experimental/GlobalHeader/index.tsx [hunk 2]
```diff
@@ -112,3 +120,8 @@ export const GlobalHeader: React.FC<GlobalHeaderProps> = (props) => (
     </Box>
   </>
 );
+
+export function useGlobalHeaderHeight(): number {
```
Comment (MattSchiller) [inline packages/gamut-labs/src/experimental/GlobalHeader/index.tsx:124]: Nice! My only recommendation is that this hook be moved into a separate file `./hooks.tsx` or whatever, to ensure you have a clean dependency graph for future callers.

Well done here!

Comment (christian-dinh) [reply packages/gamut-labs/src/experimental/GlobalHeader/index.tsx:124]: ty! No problems with making this change, just a few tradeoffs to bring up before I do (as it was the first thing I tried):

the [index file](https://github.com/Codecademy/client-modules/blob/main/packages/gamut-labs/src/experimental/index.ts) has a convention of `export * from`, which is handy because you don't have to remember to add anything to it when you export a new function/value from an already exported file. Moving the hook to a dedicated `hooks` file raises the question of where the `desktopHeight`, `mobileHeight`, and `breakpoint` consts should live, as they need to be shared between the hook file and the component file. They can't really live in either file due to the `export * from` convention (we don't really need or want them exposed to dependents), so that pretty much requires either creating a `consts` file (which imo is a lot of indirection for something so simple) or breaking the `export * from` convention and keeping them in either the hooks or component file. Likely not a huge deal either way though, 

I haven't gone too deep into our webpack configs, but re: the dependency graph I think this should still be just as tree-shakable as-is, though tbf tree-shaking is... unpredictable at best in the node_modules ecosystem. Importing the hook within `gamut-labs` would be a little awkward, but I would be surprised if any components ever did that (famous last words)

Comment (christian-dinh) [reply packages/gamut-labs/src/experimental/GlobalHeader/index.tsx:124]: omg sorry that was so many words, the differences are all really trivial but the description was not 🙈

Comment (MattSchiller) [reply packages/gamut-labs/src/experimental/GlobalHeader/index.tsx:124]: lol, no worries. Why not just _not_ export the variables for the heights, and have the header use this new shiny hook?

Then the `height` prop for both `<Box />` components are the same, `appHeaderHeight.toString()` and you don't (and shouldn't) need access to knowing that these values were ever actually different. That implementation detail is fully encapsulated within the hook.

Comment (MattSchiller) [reply packages/gamut-labs/src/experimental/GlobalHeader/index.tsx:124]: Unrelated, I think it's a bit strange that we virtually render both desktop _and_ mobile versions of the header and only show one of them at a time. I wonder if that's more performant in this case (toggling whole branches of the DOM than removing/appending the correct-for-viewport DOM branch)

Comment (christian-dinh) [reply packages/gamut-labs/src/experimental/GlobalHeader/index.tsx:124]: ah yeah so there are performance implications to using this hook in the header, as it calls `useBreakpointAtOrAbove`. The desktop+mobile side-by-siding is a perf gain over `useBreakpoint` since the latter uses js and resize handlers and the former is pure css. For the same reason we're side-by-siding, I don't think we want to use this hook in the header itself (although that would probably be the nicest possible version of this code). Discussion here: https://github.com/Codecademy/client-modules/pull/1417#discussion_r577994902

the perf implications are not super severe, but the header is on almost every page so it is probably worth avoiding. I had the exact same question re: the side-by-side so I think at this point it deserves a comment lol

Comment (MattSchiller) [reply packages/gamut-labs/src/experimental/GlobalHeader/index.tsx:124]: Yeah, +1 to the comment. I'm generally fine with leaving the hook in this file, I'm just wary of possibly circular dependencies that'll be non-obvious to future devs. I think it's slightly less-bad to export these two values to be used in these two places (and possibly, incorrectly in other places), than bake in a harder-to-find circular dependency issue.

But I leave it to you. You seem to have a great handle on the matter!

Comment (christian-dinh) [reply packages/gamut-labs/src/experimental/GlobalHeader/index.tsx:124]: safety first! I'll make the change, add a test, and post the PR for reals. ty for the discussion!

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@10.0.12-alpha.8b4c41.0<br/>@codecademy/styleguide@27.2.2-alpha.8b4c41.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://602efb7360a0a40d40f60bd7--gamut-preview.netlify.app>https://602efb7360a0a40d40f60bd7--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/602efb7360a0a40d40f60bd7>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@10.0.14-alpha.1698df.0<br/>@codecademy/styleguide@29.0.1-alpha.1698df.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://602efda6b0e35200eb236cf6--gamut-preview.netlify.app>https://602efda6b0e35200eb236cf6--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/602efda6b0e35200eb236cf6>Deploy Logs</a></p>

File: packages/gamut-labs/src/experimental/GlobalHeader/index.tsx (hunk 2 above)
Comment (rogervera) [reply packages/gamut-labs/src/experimental/GlobalHeader/index.tsx]: This was a great discussion. ^ 🥇 🥇 
The circular dependency possibility is something I wasn't even considering. Thank you both for explaining your points so clearly. I learn'd.

The current state of the PR LGTM. Love the doc entry explaining it.

File: packages/gamut-labs/src/experimental/GlobalHeader/index.tsx [hunk 3]
```diff
@@ -93,14 +94,20 @@ const getMobileAppHeaderItems = (
   }
 };
 
+// For performance reasons, this component renders the mobile and desktop versions
+// of itself simultaneously and switches between them using css (instead of using
+// any of the useBreakpoint hooks, which rely on js for the same behavior).
```
Comment (MattSchiller) [inline packages/gamut-labs/src/experimental/GlobalHeader/index.tsx:99]: 🎉

**Review by rogervera — APPROVED** [review]
LGTM. Thanks Chris for taking the time to work out and comminucate re: the best solution.

It was discussed in this thread:
https://github.com/Codecademy/client-modules/pull/1417/files#r577980940

I too am curious about the performance impact `useBreakpointAtOrAbove` does have vs rendering both versions and using css to toggle. 

If the overhead is unnecessary given the need, I wonder if we should audit and look into just replacing current use of `useBreakpointAtOrAbove` (and marking as deprecated) across the monolith/static-sites and replace it with CSS. I think in most cases, it's just for this `mobile v desktop view` need on a single component.

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
Err, what about `react-anchor-link-smooth-scroll`? Can you show an example in the monolith of needing this JS hook?

**Comment by JoshuaKGoldberg** [comment]
@rogervera yeah that was my original concern with the hooks and why I was so dubious about their need 😄 . https://codecademy.atlassian.net/browse/TR2-394 -> https://codecademy.atlassian.net/browse/REACH-475 

Conclusion: most of the time, these hooks are not a good idea to use in components that are visible on page load.

**Comment by christian-dinh** [comment]
@JoshuaKGoldberg not the monolith, but static sites:

https://github.com/codecademy-engineering/static-sites/blob/main/src/templates/LandingPage/helpers/index.tsx#L101-L107

would help us get rid of the hardcoded `80`, which is incorrect on mobile displays on top of being mysterious and fragile 👻. I think we would still need this value to do the same thing with `react-anchor-link-smooth-scroll` since you gotta set the `offset` prop

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@10.1.1-alpha.456b10.0<br/>@codecademy/styleguide@29.1.1-alpha.456b10.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://603016fe98d6e81a82af131d--gamut-preview.netlify.app>https://603016fe98d6e81a82af131d--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/603016fe98d6e81a82af131d>Deploy Logs</a></p>

File: packages/gamut-labs/src/experimental/GlobalHeader/hooks.tsx [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import { useBreakpointAtOrAbove } from '../../lib/breakpointHooks';
+import { breakpoint, desktopHeight, mobileHeight } from './consts';
+
+export function useGlobalHeaderHeight(): number {
+  const isDesktop = useBreakpointAtOrAbove(breakpoint);
```
Comment (judahtanthony) [inline packages/gamut-labs/src/experimental/GlobalHeader/hooks.tsx:5]: One thing to keep an eye out for.  That SSR version of the page doesn't have width info, so this may return ... unexpected results.

File: packages/gamut-labs/src/experimental/GlobalHeader/index.tsx [hunk 5]
```diff
@@ -93,14 +94,20 @@ const getMobileAppHeaderItems = (
...
 export const GlobalHeader: React.FC<GlobalHeaderProps> = (props) => (
   <>
-    <Box display={{ base: 'none', md: 'block' }} height="80">
+    <Box
+      display={{ base: 'none', [breakpoint]: 'block' }}
+      height={desktopHeight.toString()}
```
Comment (JoshuaKGoldberg) [inline packages/gamut-labs/src/experimental/GlobalHeader/index.tsx:104]: @codecaaron I wonder if we could expand the types for `height` and `width` to allow `number | string`, so folks didn't have to manually `toString()`?

File: packages/styleguide/stories/Labs/Experimental/Molecules/GlobalHeader.stories.mdx [hunk 6]
```diff
@@ -294,3 +294,16 @@ Check the [Navigation How-To Guide](https://docs.google.com/presentation/d/1YHG6
 1. Add the appropriate item to the `GlobalHeaderItems.tsx` file, including the necessary properities like `id`, `href`, `text`, `type`, etc.
 2. Update `GlobalHeaderVariants.tsx` to include the new item for the user roles that need it.
 3. Update tests in `GlobalHeader-test.tsx`.
+
+### How to Access GlobalHeader's height
```
Comment (JoshuaKGoldberg) [inline packages/styleguide/stories/Labs/Experimental/Molecules/GlobalHeader.stories.mdx:298]: @christian-dinh about the `resize` situation: are there situations other than scrolling to a specific element that would care about the header height? I wonder if  we could avoid the fragile number variables by using a `ref` after all, but exposing a `scrollPastHeader` function as part of a React Context instead of a hook...

File: packages/styleguide/stories/Labs/Experimental/Molecules/GlobalHeader.stories.mdx [hunk 7]
```diff
@@ -294,3 +294,16 @@ Check the [Navigation How-To Guide](https://docs.google.com/presentation/d/1YHG6
...
+You might be implementing an interaction that needs to be aware of the `GlobalHeader` height (for example when using [window.scrollTo](https://developer.mozilla.org/en-US/docs/Web/API/Window/scrollTo)). `gamut-labs` exports a `useGlobalHeaderHeight` hook for this purpose:
+
+```jsx
+import { useGlobalHeaderHeight } from 'gamut-labs';
+
+function MyComponent() {
+  const height = useGlobalHeaderHeight();
+  ...
+}
+```
```
Comment (JoshuaKGoldberg) [inline packages/styleguide/stories/Labs/Experimental/Molecules/GlobalHeader.stories.mdx:309]: Could you add a strong disclaimer about the dangers of using this in SSR-touched code paths?

File: packages/gamut-labs/src/experimental/GlobalHeader/index.tsx (hunk 5 above)
Comment (codecaaron) [reply packages/gamut-labs/src/experimental/GlobalHeader/index.tsx:104]: Sure!  It technically will already handle numbers with no units cleanly so it should just be a matter of updating the default types

File: packages/styleguide/stories/Labs/Experimental/Molecules/GlobalHeader.stories.mdx (hunk 6 above)
Comment (aresnik11) [reply packages/styleguide/stories/Labs/Experimental/Molecules/GlobalHeader.stories.mdx:298]: I think this could be used for the `SkipToContent` targets which need to offset the sticky header height. The monolith currently has a `headerHeight` variable set to `4rem` [here](https://github.com/codecademy-engineering/Codecademy/blob/develop/webpack/assets/styles/variables.ts#L2) and utilizes it for a negative margin top for the target [here](https://github.com/codecademy-engineering/Codecademy/blob/develop/webpack/assets/components/SkipToContentTarget/index.tsx#L11). Static sites is currently just hard coding the `-4rem` [here](https://github.com/codecademy-engineering/static-sites/blob/main/src/components/shared/PageContainer/index.tsx#L26).

Comment (MattSchiller) [reply packages/styleguide/stories/Labs/Experimental/Molecules/GlobalHeader.stories.mdx:298]: It's also possible that we would want to forego this pathway entirely in favor of css variables if we _only_ want this information for CSS rules.

Like, we can include a base sass file with something like:
```
:root {
  --global-header-height: 64px;
}

@media screen and (max-width: 800) { // or whatever our mobile breakpoint is {
  :root {
    --global-header-height: 80px;
    }
  }
  ```
  
And then we don't expect our components to receive this information through React (JS) at all. We just update their sass stylings to use these variables directly.

_If_ we only want these values for css purposes, then I think this accomplishes the best of both worlds (aka, The Hannah Montana solution), where we _never_ have the flash of incorrect styling because SSR doesn't have a width value. It's always rendered correctly on the first pass from CSS.

Thoughts?

**Review by JoshuaKGoldberg — DISMISSED** [review]
The CSS variable discussion is interesting, but I don't have any other ideas so approving anyway 😄

File: packages/styleguide/stories/Labs/Experimental/Molecules/GlobalHeader.stories.mdx (hunk 6 above)
Comment (christian-dinh) [reply packages/styleguide/stories/Labs/Experimental/Molecules/GlobalHeader.stories.mdx:298]: that's a pretty strong proposal, and actually if we expose a ref on the header we should still be able to access the value in js:
```js
getComputedStyle(headerRef.current).getPropertyValue('--global-header-height');
```
I do think that js visibility is a requirement for header-clearing scroll targets (since `scroll-margin-top` is not supported in safari), but this feels pretty good to me if it works.

Comment (christian-dinh) [reply packages/styleguide/stories/Labs/Experimental/Molecules/GlobalHeader.stories.mdx:298]: or yeah put it on :root and then we don't need a ref

Comment (christian-dinh) [reply packages/styleguide/stories/Labs/Experimental/Molecules/GlobalHeader.stories.mdx:298]: ok so this totally works, but some caveats for discussion:
- we can add `--global-header-height` to `:root` and access it via an exported helper function, but that function is not unit-testable because [jsdom does not support the 'c' in 'css' for variables](https://github.com/jsdom/cssstyle/pull/111)
- we can add `--global-header-height` to the `GlobalHeader` itself and the helper function _should_ (I haven't tried yet) be unit-testable (and would probably exist as a [getter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/get) on `GlobalHeader`), but we will need a ref on the header to access the value elsewhere. This will be slightly painful in static-sites for example, where we would likely end up shoving it into context
- neither method works with hooks as the helper function is inherently imperative, however I imagine any non-css code that would rely on the header height will also be imperative so imo this is fine

tl;dr there's an easy jank way and an annoying but more correcter way. I'm personally a big fan of both compared to the previous iterations though!

Comment (MattSchiller) [reply packages/styleguide/stories/Labs/Experimental/Molecules/GlobalHeader.stories.mdx:298]: Oh, interesting. Do you have some sample code to check out? I was hoping this would be simpler, covering the css use-cases, but I understand the need for JS-availability of the values.

As I understand the aspects of this puzzle:
A) We'd prefer to leverage CSS to determine the correct value wherever possible, instead of using JS to calc the screen width, as this prevents flashing incorrect styling in SSR
B) We still want these values to be accessible from JS for other things
C) Keep everything defined in one place, testable, and limited in what it exposes

Would it be useful to do the following (and maybe this is what you're describing that I'm just not getting atm):
1) Snippet of CSS code above, but converted into rendering a style tag that sets the variables higher in the tree than the header itself, in a `<CssVariables />` or something:
2) Create a hook that tries to access these variables, but defaults to mobile or desktop or whatever in the case that it cannot find it (which is our best case for JS-based SSR rendering for these values anyway):
```
const globalHeaderHeightKey = '--global-header-height';
const desktopHeaderHeight = 80; // Note: not exported anymore
const mobileHeaderHight = 64;

// Any css-based needs can get these values from var(--global-header-height)
// NB: This is just some untested hacky code
export const CssVariables = () => (
  <style>{"\
    :root {
      ${globalHeaderHeightKey}:${mobileHeaderHight}px;
    }

    @media screen and (max-width: 800) { // or whatever our mobile breakpoint is {
        :root {
          ${globalHeaderHeightKey}: ${desktopHeaderHeight}px;
        }
     }
   "}</style>
)

// Any JS-based needs can use this hook instead
export const useGlobalHeaderHeight = (): number => 
  getComputedStyle(document?.documentElement)
    .getPropertyValue(globalHeaderHeightKey) || desktopHeaderHeight;
```

In this way, we could create a central repository in the DOM tree (ideally _very high_) that any variables like this can just be added to and then both CSS and JS have access to it (either via css vars or react hooks).

And we could test it, by just confirming that the `<CssVariables />` has the correct snapshot, without having to actually rely on confirming the CSS cascading.

Thoughts? I'm crazy, right?

Comment (christian-dinh) [reply packages/styleguide/stories/Labs/Experimental/Molecules/GlobalHeader.stories.mdx:298]: Your understanding of the puzzle is correct, although just to add to `A` we also want to avoid js calcs on resize for performance reasons

re: #1, that's actually exactly what I'm doing atm (via [emotion globals](https://emotion.sh/docs/globals)) 

re: #2, the issue I've run into with hooks in this case is that afaik the hook value cannot update on page resize without a js listener. The most performant solution I've found is an imperative helper function that is called when we need the value, which should be fine because the main use-cases are smooth-scrolling and skip nav which are imperative actions.

I'll push what I have currently! It works but is not easily testable because on top of cascade jsdom does not seem to implement the `:root` pseudoselector </3 which is why I think applying the variables to `GlobalHeader` directly would be more testable but less ergonomic (requires ref) but more react-ive(?) (uses a ref for imperative things and avoids globals). idk, tradeoffs to be made for sure. If there's a clever way to unit test this as-is in absence of `:root` I would be 100% happy with that solution.

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@10.1.1-alpha.41ff69.0<br/>@codecademy/styleguide@29.1.1-alpha.41ff69.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://603534ac748b1d00a30dd7fc--gamut-preview.netlify.app>https://603534ac748b1d00a30dd7fc--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/603534ac748b1d00a30dd7fc>Deploy Logs</a></p>

File: packages/gamut-labs/src/experimental/GlobalHeader/hooks.tsx [hunk 8]
```diff
@@ -0,0 +1,8 @@
+export function getGlobalHeaderHeight(): number {
+  return parseInt(
+    getComputedStyle(document.documentElement).getPropertyValue(
+      '--global-header-height'
+    ),
+    10
+  );
+}
```
Comment (MattSchiller) [inline packages/gamut-labs/src/experimental/GlobalHeader/hooks.tsx:8]: Nit: This method is no longer a hook, and should probably be renamed `/helpers` or something

File: packages/styleguide/stories/Labs/Experimental/Molecules/GlobalHeader.stories.mdx (hunk 6 above)
Comment (MattSchiller) [reply packages/styleguide/stories/Labs/Experimental/Molecules/GlobalHeader.stories.mdx:298]: Gotcha, I get it now (and learned about emotion globals!). Thanks for the explanation!

Would a snapshot of `<GlobalHeader />`'s `<Global />` not be sufficient to confirm that it's providing the values we're looking for? Like, we don't usually test to confirm that CSS is being applied in the way we expect it to, do we? Would a cypress test be helpful if we _really_ want to confirm that they're being applied correctly? 

And for the helper, that should be testable, right?

I'm generally inclined to keep these variables at the `:root` level (for ease of access in css) over using a `ref`, especially since it seems like we're going to be looking to use this in JS in non-react-ive patterns (imperatively, as you pointed out). 

I think what you've got here is especially awesome already. Excited to have this kind of flexibility in our repo!

Comment (christian-dinh) [reply packages/styleguide/stories/Labs/Experimental/Molecules/GlobalHeader.stories.mdx:298]: Totally fine with keeping it at `:root` for the reasons you mentioned. I was actually specifically having trouble testing the helper, but I think I might have just had a Big Brain Moment™ 🤯  stand by

Comment (christian-dinh) [reply packages/styleguide/stories/Labs/Experimental/Molecules/GlobalHeader.stories.mdx:298]: ok I switched up what the test was testing and I think this feels pretty good now. Still need to update docs but lmk what u think.

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@10.1.1-alpha.468fb4.0<br/>@codecademy/styleguide@29.1.1-alpha.468fb4.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60357deee5a5d8007d06db80--gamut-preview.netlify.app>https://60357deee5a5d8007d06db80--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60357deee5a5d8007d06db80>Deploy Logs</a></p>

File: packages/styleguide/stories/Labs/Experimental/Molecules/GlobalHeader.stories.mdx (hunk 6 above)
Comment (codecaaron) [reply packages/styleguide/stories/Labs/Experimental/Molecules/GlobalHeader.stories.mdx:298]: Sorry my computer has been out of commission since the weekend.  Finally caught up on setting up new on.

I'll take a look at these comments tonight.

File: packages/gamut-labs/src/experimental/GlobalHeader/util.tsx [hunk 9]
```diff
@@ -0,0 +1,10 @@
+import { headerHeightVarName } from './consts';
+
+export function getGlobalHeaderHeight(): number {
+  return parseInt(
+    getComputedStyle(document.documentElement).getPropertyValue(
```
Comment (MattSchiller) [inline packages/gamut-labs/src/experimental/GlobalHeader/util.tsx:5]: nit: Does this need the optional chaining on `document` to work on SSR? Or does it fail gracefully as-is? I dunno myself, just wondering!

File: packages/styleguide/stories/Labs/Experimental/Molecules/GlobalHeader.stories.mdx (hunk 6 above)
Comment (MattSchiller) [reply packages/styleguide/stories/Labs/Experimental/Molecules/GlobalHeader.stories.mdx:298]: @christian-dinh Thank you for all of this! Not only do I think this is some awesome code, but I've loved the conversation about it.

I left only a single nit/question, otherwise, to whatever extent my approval matters, I think this is some Grade A code right here.

File: packages/styleguide/stories/Labs/Experimental/Molecules/GlobalHeader.stories.mdx (hunk 7 above)
Comment (saghdaey) [reply packages/styleguide/stories/Labs/Experimental/Molecules/GlobalHeader.stories.mdx:309]: just want to flag that the anonymous landing page header will be using the GlobalHeader, and I believe that page is rendered server side!

Comment (saghdaey) [reply packages/styleguide/stories/Labs/Experimental/Molecules/GlobalHeader.stories.mdx:309]: just want to flag that the anonymous landing page header (in the monolith) will be using the GlobalHeader

Comment (saghdaey) [reply packages/styleguide/stories/Labs/Experimental/Molecules/GlobalHeader.stories.mdx:309]: I have a draft PR up now for the global header monolith migration

File: packages/gamut-labs/src/experimental/GlobalHeader/util.tsx (hunk 9 above)
Comment (JoshuaKGoldberg) [reply packages/gamut-labs/src/experimental/GlobalHeader/util.tsx:5]: `document` won't exist in SSR in the monolith anymore. So we'll want to put this on a preview server for sure. 😄

Comment (christian-dinh) [reply packages/gamut-labs/src/experimental/GlobalHeader/util.tsx:5]: given the use case I don't see when this function would be called during SSR, though if it is would it be helpful to log a console warning?

Comment (JoshuaKGoldberg) [reply packages/gamut-labs/src/experimental/GlobalHeader/util.tsx:5]: I think I like it as -- it should aggressively crash if used in a component's initial render, as a way to tell us to not do that thing 😄

Comment (MattSchiller) [reply packages/gamut-labs/src/experimental/GlobalHeader/util.tsx:5]: That's a great idea, yea!

**Comment by christian-dinh** [comment]
closing cause https://github.com/Codecademy/client-modules/pull/1444 does all this and more 🎉
